src/components/select/SelectItemCommons.tsx

```diff
@@ -42,13 +42,30 @@ function withSelectItemWrapper<P extends SelectOption>(
   optionType: SelectOptionType,
 ) {
   const Wrapper = forwardRef<HTMLDivElement, P>((props, ref) => {
-    const { helpText, label, prefix, suffix, value, ...rest } = props;
+    const { helpText, label, prefix, selected, suffix, value, ...mantineAndCustomProps } = props;
+
+    const {
+      'aria-selected': _ariaSelected,
+      className,
+      'data-disabled': _dataDisabled,
+      'data-hovered': _dataHovered,
+      'data-selected': _dataSelected,
+      onMouseDown,
+      onMouseEnter,
+      tabIndex,
+      ...customProps
```

Comment:
You could rename it to `customComponentProps`, to make it more related to the custom component, and not restrict it to the custom fields in the option.

Also it would be nice to add a comment saying something like: `// remove the mantine props from the custom component props` to make it more understandable why all these extracted things are not used later one.